cli/provisionerrun_test.go

```diff
@@ -0,0 +1,103 @@
+package cli_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/cli/clitest"
+	"github.com/coder/coder/coderd/coderdtest"
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/provisioner/echo"
+	"github.com/coder/coder/pty/ptytest"
+)
+
+func TestProvisionerRun(t *testing.T) {
+	t.Parallel()
+	t.Run("Provisioner", func(t *testing.T) {
+		t.Parallel()
+		client := coderdtest.New(t, nil)
+		_ = coderdtest.CreateFirstUser(t, client)
+		provisionerResponse, err := client.CreateProvisionerDaemon(context.Background(),
+			codersdk.CreateProvisionerDaemonRequest{
+				Name: "foobar",
+			},
+		)
+		require.NoError(t, err)
+		token := provisionerResponse.AuthToken
+		require.NotNil(t, token)
+
+		doneCh := make(chan error)
+		defer func() {
+			err := <-doneCh
+			require.ErrorIs(t, err, context.Canceled, "provisioner command terminated with error")
+		}()
+
+		ctx, cancelFunc := context.WithCancel(context.Background())
```

Comment:
Consider allowing tests to timeout individually, e.g.:

```suggestion
		ctx, cancelFunc := context.WithTimeout(context.Background(), testutil.WaitLong)
```